File: 014 - Generic Stack/GenericStack/GenericStack/Stack.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GenericStack
{
    internal class Stack<T>
    {
        private List<T> Items;
        public Stack ()
        {
            Items = new List<T>();
        }
        
        public void push (T item)
        {
            Items.Add(item);
        }

        public T pop ()
        {
            if (Items.Count == 0)
            {
                throw new InvalidOperationException("Stack is Empty ");
            }
            T item = Items[Items.Count - 1];
            Items.RemoveAt(Items.Count - 1);
            return item;
        }

        public bool IsEmpty()
        {
            return Items.Count == 0;
        }

    }
}

File: DSA problems/Stack/Operations/Operations/Stack.cpp
#include "Stack.h"
#include <iostream>
using namespace std;

const int Size = 10;
class Stack
{
private:
	int stack[Size];
	int top = -1;
public:
	 
	bool IsEmpty() {
		if (top == - 1) {
			return true;
		}
		else
			return false;
	}
	bool IsFull() {
		if (top == Size - 1) {
			return true;

		}
		else
			return false;
	}

	void Push( int value) {
		if (!IsFull()) {

			stack[++top] = value;
		}
		else
			cout << "Stack overflow";
	}
	int pop() {
		if (!IsEmpty()) {
			return stack[--top];
		}
		cout << "Stack is Empty";
	}
	int peek() {
		if (!IsEmpty()) {
			return stack[top];
		}
		cout << "Stack is Empty";
	}
	void Display() {
		if (IsEmpty()) {
			cout << "Stack is Empty";
			return;
		}
		for (int i = top; i >= 0; i--) {
			cout << stack[i] << endl;
		}
	}

};

